force-app/main/default/lwc/nksSubtitles/nksSubtitles.js

```diff
@@ -0,0 +1,114 @@
+import { LightningElement, wire, api } from 'lwc';
+import { ShowToastEvent } from 'lightning/platformShowToastEvent';
+import getFileType from '@salesforce/apex/NKS_VideoPlayerCtrl.checkFileType';
+import getSubtitleLanguageLinksOnFile from '@salesforce/apex/NKS_VideoPlayerCtrl.getSubtitleLanguageLinksOnFile';
+import saveSubtitleLanguageLinks from '@salesforce/apex/NKS_VideoPlayerCtrl.saveSubtitleLanguageLinks';
+import showVideoTrackURL from '@salesforce/apex/NKS_VideoPlayerCtrl.showVideoTrackURL';
+
+export default class NksSubtitles extends LightningElement {
+    @api recordId; // The Content Document we are on
+    contentVersionId; // The related CV Id of the Content Document
+    subtitleLinks = [];
+
+    @wire(getSubtitleLanguageLinksOnFile, { videoId: '$recordId' })
+    wiredgetSubtitleLanguageLinksOnFile(result) {
+        if (result.error) {
+            console.error(result.error);
+        } else if (result.data) {
+            this.contentVersionId = result.data.Id;
+            this.subtitleLinks = this.setSubtitleLinkDataOnWire(Object.assign({}, result.data));
+        }
+    }
+
+    filetype;
+    @wire(getFileType, { recordId: '$recordId'})
+    wiredGetFileType(result) {
+        if (result.error) {
+            console.log(result.error);
+        } else if (result.data) {
+            this.filetype = result.data;
+        }
+    }
+
+    get isVideoFile() {
+        return isVideoFile(this.filetype);
+    }
+
+    get isSubtitleFile() {
+        return isSubtitleFile(this.filetype);
+    }
+    
+    videoTrackURL;
+    @wire(showVideoTrackURL, { videoId: '$recordId'})
+    wiredShowVideoTrackURL(result) {
+        if (result.error) {
+            console.log(result.error);
+        } else if (result.data) {
+            this.videoTrackURL = result.data;
+        }
+    }
+
+    setSubtitleLinkDataOnWire(recordData) {
+        const languageMap = { NKS_Subtitle_Link_Norwegian__c: 'Norsk', NKS_Subtitle_Link_English__c: 'Engelsk', NKS_Subtitle_Link_Polish__c: 'Polsk' };
+        delete recordData.Id; // Do not create another input field based on Id
+        let data = [];
+        for (const field in recordData) {
+            data.push({ language: languageMap[field], link: recordData[field], relatedField: field });
+        }
+        return data;
+    }
+
+    saveSubtitleLink() {
+        let data = {};
+        this.subtitleLinks.forEach(element => {
+            data[element.relatedField] = element.link;
+        });
+        data.Id = this.contentVersionId;
+        saveSubtitleLanguageLinks({ cvObj: data });
+        this.showSaveToast();
```

Comment:
Kjør saveToasten etter APEXen faktisk har blitt kjørt, altså i en then(), og legg på en catch for å visse error meldinger. Vi vil ikke gi veilederne (kaller vi fortsatt webinargjengen for det?) en false positiv hvis det feiler.